dts: bindings: pwm custom macro for STM32 vendor

FRASTM opened the pull request:
Change the name of the custom macro defined for the stm32 devices to fit the VND_PWM_xxx model
This is **STM32_PWM_COMPLEMENTARY**

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
dts: bindings: pwm custom macro for STM32 vendor

Change the name of the custom macro defined for the stm32 devices
to fit the VND_PWM_xxx model
Keeping old deprecated macro, though.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on include/zephyr/dt-bindings/pwm/stm32_pwm.h:
```diff
@@ -16,10 +16,10 @@
 /** PWM complementary output pin is enabled */
-#define PWM_STM32_COMPLEMENTARY	(1U << 8)
+#define STM32_PWM_COMPLEMENTARY	(1U << 8)
 
 /** @cond INTERNAL_HIDDEN */
-#define PWM_STM32_COMPLEMENTARY_MASK	0x100
+#define STM32_PWM_COMPLEMENTARY_MASK	0x100
```
Let's make change non breaking
```suggestion
#define PWM_STM32_COMPLEMENTARY_MASK	__DEPRECATED_MACRO 0x100
#define STM32_PWM_COMPLEMENTARY_MASK	0x100
```

FRASTM replied:
ok

erwango commented on include/zephyr/dt-bindings/pwm/stm32_pwm.h:
```diff
@@ -16,10 +16,10 @@
  */
 /** PWM complementary output pin is enabled */
-#define PWM_STM32_COMPLEMENTARY	(1U << 8)
+#define STM32_PWM_COMPLEMENTARY	(1U << 8)
```
Let's make change non breaking
```suggestion
#define PWM_STM32_COMPLEMENTARY	__DEPRECATED_MACRO (1U << 8)
#define STM32_PWM_COMPLEMENTARY	(1U << 8)
```

FRASTM replied:
ok

FRASTM commented:
keeping old macro for compatibility

FRASTM commented:
fixing compliance error

henrikbrixandersen commented on include/zephyr/dt-bindings/pwm/stm32_pwm.h:
```diff
@@ -16,10 +16,12 @@
  */
 /** PWM complementary output pin is enabled */
-#define PWM_STM32_COMPLEMENTARY	(1U << 8)
+#define PWM_STM32_COMPLEMENTARY	(1U << 8) __DEPRECATED_MACRO
```
`__DEPRECATED_MACRO` will not work for definitions used in devicetree.

erwango replied:
Arf, so what are the guidelines to deprecate dts bindings ?

henrikbrixandersen reviewed (changes requested):
You'll need to fix the `nucleo_l4r5zi` board DTS as well.

FRASTM committed:
boards: arm: nucleo_l45r change the STM32_PWM_COMPLEMENTARY macro

Change with the new STM32_PWM_COMPLEMENTARY macro for the nucleo_l4r5zi
board.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
Changed with @doxygen tag
The PWM_STM32_COMPLEMENTARY_MASK does not need the old style as it is internal
Use the new macro  in the DTS  of the nucleo_l4r5zi board

henrikbrixandersen commented on include/zephyr/dt-bindings/pwm/stm32_pwm.h:
```diff
@@ -16,10 +16,14 @@
+/**
+ * @deprecated Use the PWM complementary `STM32_PWM_COMPLEMENTARY` flag instead.
+ */
+#define PWM_STM32_COMPLEMENTARY	(1U << 8) __DEPRECATED_MACRO
```
You left the macro here.

FRASTM replied:
... that's like in the commit you mention
![capture](https://user-images.githubusercontent.com/51417392/213457374-6df653ba-8c0b-4f58-8cad-26852e12150b.PNG)

FRASTM replied:
removing "__DEPRECATED_MACRO" on PWM_STM32_COMPLEMENTARY

erwango reviewed (approved):
Doc build issue seems unrelated
